product_sequence/models/product_product.py

```diff
@@ -49,10 +46,7 @@ def write(self, vals):
             for product in self:
                 category_id = vals.get('categ_id', product.categ_id.id)
                 category = product_category_obj.browse(category_id)
-                sequence = category.exists() and category.sequence_id
-                if not sequence:
-                    sequence = self.env.ref(
-                        'product_sequence.seq_product_auto')
+                sequence = category.get_sequence_id()
```

Comment:
Maybe you can keep the if clause and call you function there.

Reply:
Or you can define you function like this.
`@api.model`
`def get_sequence_id(self, category):`

Reply:
done